docroot/modules/custom/foia_webform/src/Plugin/WebformHandler/FoiaEmailWebformHandler.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\foia_webform\Plugin\WebformHandler;
+
+use Drupal\file\Entity\File;
+use Drupal\webform\Plugin\WebformHandler\EmailWebformHandler;
+use Drupal\webform\WebformSubmissionInterface;
+use Drupal\node\Entity\Node;
+
+/**
+ * Emails a webform submission.
+ *
+ * @Webformhandler(
+ *   id = "foia_email",
+ *   label = @Translation("FOIA email"),
+ *   category = @Translation("Notification"),
+ *   description = @Translation("Sends a webform submission to appropriate the agency."),
+ *   cardinality = \Drupal\webform\WebformHandlerInterface::CARDINALITY_UNLIMITED,
+ *   results = \Drupal\webform\WebformHandlerInterface::RESULTS_PROCESSED,
+ * )
+ */
+class FoiaEmailWebformHandler extends EmailWebformHandler {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function sendMessage(WebformSubmissionInterface $webform_submission, array $message) {
+    // Get form submissions.
+    $data = $webform_submission->getData();
+    $error_message = NULL;
+    $context = [];
+
+    // If there is a file attachment, get the file URL.
+    if (isset($data['attachments_supporting_documentation'])) {
+      $file = File::load($data['attachments_supporting_documentation']);
+      $file_url = file_create_url($file->getFileUri());
+      $data['attachments_supporting_documentation'] = $file_url;
+    }
+
+    // Format the submission values as CSV.
+    $submissions = $this->arrayToString($data);
+
+    // Append CSV string to the message body.
+    $newline = ($message['html']) ? '<br/>' : "\n";
+    $text = t('The submission data is reproduced below in CSV format for your convenience.');
+    $message['body'] .= $text . $newline;
+    $message['body'] .= '--------------------------' . $newline;
+    $message['body'] .= $submissions;
+
+    // Look up the agency component.
+    $form = $webform_submission->getWebform();
+    $form_id = $form->getOriginalId();
+    $agency_component = $this->lookupComponent($form_id);
+
+    // If we have an Agency Component, get the Submission Email value.
+    if ($agency_component) {
+      $to_email = $agency_component->get('field_submission_email')->getValue();
+
+      if (!empty($to_email)) {
+        $message['to_mail'] = $to_email[0]['value'];
+      }
+      // If we don't have a Submission Email value log an error.
+      else {
+        $error_message = 'No Submission Email: Unable to send email for %title';
+        $context = [
+          '%title' => $agency_component->getTitle(),
+          'link' => $agency_component->toLink($this->t('Edit Component'), 'edit-form')->toString(),
+        ];
+      }
+    }
+    // If there isn't an associated Agency Component log an error.
+    else {
+      $error_message = 'Unassociated form: The form, %title, is not associated with an Agency Component.';
```

Comment:
Random thoughts:
1) I'm wondering if this should actually be considered an error. If we ever have a webform on the site that isn't related to the portal, it may not need to be associated with a component.
2) It would be nice if we only added this email handler to webforms associated with components (in case we have webforms that don't need to send out e-mails), but that's not very easy with the current architecture.

Again, these are random thoughts and discussion points, nothing to kick the ticket back for.